float32 x_percentage
float32 y_percentage